src/mqt/predictor/rl/Predictor.py

```diff
@@ -32,67 +32,61 @@ def compile_as_predicted(
+            env = rl.PredictorEnv(rew)
             obs = env.reset(file)
+
             qc = env.state
             start_time = time.time()
             while True:
                 action_masks = get_action_masks(env)
                 action, _states = model.predict(obs, action_masks=action_masks)
                 action = int(action)
                 obs, reward_val, done, info = env.step(action)
+
                 if done:
                     duration = time.time() - start_time
                     if rew == "fidelity":
-                        RL_fid = np.round(
-                            reward.expected_fidelity(env.state, env.device), 2
-                        )
+                        RL_fid = reward.expected_fidelity(env.state, env.device)
                         RL_fid_time = np.round(duration, 2)
-                        RL_fid_crit_depth = np.round(reward.crit_depth(env.state), 2)
-                        RL_fid_parallelism = np.round(reward.parallelism(env.state), 2)
-                    elif rew == "parallelism":
-                        RL_parallelism = np.round(reward.parallelism(env.state), 2)
-                        RL_parallelism_time = np.round(duration, 2)
-                        RL_parallelism_fid = np.round(
-                            reward.expected_fidelity(env.state, env.device), 2
-                        )
-                        RL_parallelism_crit_depth = np.round(
-                            reward.crit_depth(env.state), 2
+                        RL_fid_crit_depth = reward.crit_depth(env.state)
+                        RL_fid_gate_ratio = reward.gate_ratio(env.state)
+                    elif rew == "gates":
+                        RL_gate_ratio = reward.gate_ratio(env.state)
+                        RL_gate_ratio_time = np.round(duration, 2)
+                        RL_gate_ratio_fid = reward.expected_fidelity(
+                            env.state, env.device
                         )
+                        RL_gate_ratio_crit_depth = reward.crit_depth(env.state)
                     elif rew == "critical_depth":
-                        RL_crit_depth = np.round(reward.crit_depth(env.state), 2)
+                        RL_crit_depth = reward.crit_depth(env.state)
                         RL_crit_depth_time = np.round(duration, 2)
-                        RL_crit_depth_fid = np.round(
-                            reward.expected_fidelity(env.state, env.device), 2
-                        )
-                        RL_crit_depth_parallelism = np.round(
-                            reward.parallelism(env.state), 2
+                        RL_crit_depth_fid = reward.expected_fidelity(
+                            env.state, env.device
                         )
+                        RL_crit_depth_gate_ratio = reward.gate_ratio(env.state)
+                    elif rew == "mix":
+                        RL_mix = reward.mix(env.state, env.device)
```

Comment:
I added that also for the mix reward. First, I thought it is just another reward we tested and thus, I thought it might not be necessary to integrated it also in a similar fashion as the others for our comparison that the optimization goal is adaptable. But now I think it does and implemented it together with the new result structure.